cmd/argocd/commands/account.go

```diff
@@ -80,14 +82,27 @@ func NewAccountUpdatePasswordCommand(clientOpts *argocdclient.ClientOptions) *co
 				errors.CheckError(err)
 			}
 
+			//Need to validate password complexity with regular expression
+			settingsConn, settingsIf, err := acdClient.NewSettingsClient()
+			defer io.Close(settingsConn)
+			errors.CheckError(err)
+			argoSettings, err := settingsIf.Get(context.Background(), &settingspkg.SettingsQuery{})
+			errors.CheckError(err)
+			passwordPattern := argoSettings.PasswordPattern
+			var validPasswordRegexp = regexp.MustCompile(passwordPattern)
+			if !validPasswordRegexp.Match([]byte(newPassword)) {
```

Comment:
The password must be validated on server side, not CLI. Please move the validation logic to https://github.com/argoproj/argo-cd/blob/1d3d03f5d3f62755ae90730121e75124c88340fe/server/account/account.go#L79

Note: the `passwordPattern` field in settings API is still required. UI will use it later to perform client side validation.